### PR TITLE
don't call connect if mail host is specified

### PR DESCRIPTION
**Pull request by smoors**
as explained in the smtplib docs at https://docs.python.org/3/library/smtplib.html#smtplib.SMTP.connect
"This method is automatically invoked by the constructor if a host is specified during instantiation."

~~the code in mail.py is written in such a way that the mail_host must always be specified, so `connect` is always called automatically.~~

calling `connect` manually (i.e. a second time) causes the process to hang without sending any mails.

### CODE REVIEW

File: lib/vsc/utils/mail.py [hunk 1]
```diff
@@ -115,8 +115,6 @@ def _connect(self):
             context = ssl.create_default_context()
             s.starttls(context=context)
             logging.debug("Started TLS connection")
-        else:
-            s.connect()
```
Comment (itkovian) [inline lib/vsc/utils/mail.py:119]: I distinctly recall this was needed at some point.

Comment (smoors) [reply lib/vsc/utils/mail.py:119]: @itkovian it's indeed still needed if the mail host is not specified.
I was wrong thinking that it must always be specified.
fixed in https://github.com/hpcugent/vsc-base/pull/331/commits/1b9ff71f551fd877f7673fb4f99f5228e315ecf0

**Comment by smoors** [comment]
@itkovian any idea why jenkins fails?

**Comment by itkovian** [comment]
@smoors 
```
FAIL: test_timeout (test.run.TestRun)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/var/lib/jenkins/jobs/hpcugent_github.com/jobs/vsc-base/branches/PR-331/workspace/test/run.py", line 295, in test_timeout
    do_test(False)
  File "/var/lib/jenkins/jobs/hpcugent_github.com/jobs/vsc-base/branches/PR-331/workspace/test/run.py", line 273, in do_test
    self.assertFalse(check_pid(pids[depth]), "main depth=%s pid (pids %s) is killed by timeout" % (depth, pids,))
AssertionError: True is not false : main depth=2 pid (pids [13852, 13850, 13849]) is killed by timeout
```

**Pull request merged**